### ISSUE TITLE
The C API lacks a binary safe leveldb_get() function.

### ISSUE THREAD

**Issue by cmumford** [google/leveldb#156]
Original [issue 150](https://code.google.com/p/leveldb/issues/detail?id=150) created by firealwaysworks on 2013-02-26T21:24:51.000Z:

<b>What steps will reproduce the problem?</b>
My C program that is using LevelDB for storing binary blobs and I ran into the following problem with the LevelDB C API:

1)Leveldb_put() allows the programmer to pass a char\* and a length for both the &quot;key&quot; and &quot;val&quot;,  which allows the programmer to store binary data,  which may contain null bytes.

2)leveldb_get() returns a char\* without a size parameter...  Digging into ./leveldb/c.cc you can see that it is using a &quot;std:string&quot; to return this char*...  Not only is this not binary safe,  but std::string?  Having an efficient slice, like the one provided in the C++ api would be much preferred.

So in short; using the C API its impossible to put() items without ever being able to get() them back...

The solution is to provide a leveldb_get_binary()  which allows the programmer to pass in a reference to char\* and size param.  The call convention should be consistent with the leveldb_put() function.

<b>What is the expected output? What do you see instead?</b>
I expected to obtain an unmolested blob of binary,  I got a null terminated string. 

<b>What version of the product are you using? On what operating system?</b>
libleveldb-dev:amd64 0+20120530.gitdd0d562-2 
installed with apt,  under Ubuntu.
